visualization/app/codeCharta/ui/copyToClipboardButton/util/getFilenamesWithHighestMetrics.ts

```diff
@@ -0,0 +1,41 @@
+import { hierarchy } from "d3-hierarchy"
+import { CodeMapNode, NodeType } from "../../../codeCharta.model"
+
+export type FileToValue = { name: string; value: number }
+
+const MAX_ENTRIES = 10
+
+export function getFilenamesWithHighestMetrics(node: Pick<CodeMapNode, "children" | "type" | "attributes" | "name">) {
+	const fileToValueByAttributes = new Map<string, FileToValue[]>()
+	for (const { data } of hierarchy(node)) {
+		if (data.type === NodeType.FILE && data.attributes) {
+			for (const [key, value] of Object.entries(data.attributes)) {
+				updateAttributeMap(key, value, data.name, fileToValueByAttributes)
+			}
+		}
+	}
+	return fileToValueByAttributes
+}
+
+export function updateAttributeMap(key: string, value: number, fileName: string, map: Map<string, FileToValue[]>) {
+	const keyIsNew = !map.has(key)
+	if (keyIsNew) {
+		map.set(key, [{ name: fileName, value }])
+		return
+	}
+
+	const newPair = { name: fileName, value }
+	const previousValues = map.get(key)
+
+	// NOTE: Perfomance can be improved by 10% to 25% by optimizing insertion algorithm for full arrays. See getFilenamesWithHighestMetrics.md-file for details
```

Comment:
There is a typo in "Perfomance". Anyway, see comment above. Or feel free to implement and extract the more performant way, if it was already implemented.

Reply:
Resolved with commit [4045391](https://github.com/MaibornWolff/codecharta/pull/2942/commits/40453916f006e10ae233021e376bd307576a0f54)